dotnet/src/SemanticKernel.Abstractions/AI/ChatCompletion/ChatHistory.cs

```diff
@@ -124,6 +124,36 @@ public void AddRange(IEnumerable<ChatMessageContent> items)
         this._messages.AddRange(items);
     }
 
+    /// <summary>
+    /// Add a message to the chat history at the end of the streamed message
+    /// </summary>
+    /// <typeparam name="T">Specific chat message content type</typeparam>
+    /// <param name="streamingMessageContents"><see cref="IAsyncEnumerator{T}"/> list of streaming message contents</param>
+    /// <returns>Returns the original streaming results with some message processing</returns>
+    /// <exception cref="InvalidOperationException"></exception>
+    public async IAsyncEnumerable<T> AddStreamingMessageAsync<T>(IAsyncEnumerable<T> streamingMessageContents) where T : StreamingChatMessageContent
+    {
+        List<T> messageContents = new();
+        AuthorRole? role = null;
+        StringBuilder messageContent = new();
+
+        await foreach (var chatMessage in streamingMessageContents)
+        {
+            messageContent.Append(chatMessage.Content);
+            messageContents.Add(chatMessage);
+
+            // Is aways expected to have at least one chunk with the role provided from a streaming message
+            role ??= chatMessage.Role;
+
+            yield return chatMessage;
+        }
+
+        if (messageContents.Count != 0)
```

Comment:
How likely is this?  If it's expected to not be super rare, we should consider lazily-initializing messageContents and messageContent so we don't pay for those allocations if there are no messages.  If it's expected to be very rare (which I assume is the case), then just proactively allocating those is fine.

Reply:
I think is super rare getting streaming without contents, I'm adding this as a safety check.